Scripts/Levels/Level Main.gd

```diff
@@ -141,18 +160,29 @@ func process_actions(actions):
 
 var turn_end_actions = []
 var turn_start_actions = []
+# Reserved for moves that only activate for the player who casted them
+var player_turn_actions = []
 
-func register_repeated_action(object, method, num_times, siignal):
+# Format for actions
+# [Organism, Ability_Name, Duration, Target Player]
+func register_repeated_action(object, method, num_times, player, siignal):
 	if siignal == "turn_end":
-		turn_end_actions.append([object, method, num_times])
+		turn_end_actions.append([object, method, num_times, null])
 	elif siignal == "turn_start":
-		turn_start_actions.append([object, method, num_times])
+		turn_start_actions.append([object, method, num_times, null])
+	elif siignal == "player_turn":
+		player_turn_actions.append([object, method, num_times, player])
 
 var move_icon_active = load("res://Assets/UI/Player/Game_Player_Moves_Icon_Active.png")
 var move_icon_used = load("res://Assets/UI/Player/Game_Player_Moves_Icon_Used.png")
 onready var move_icons = $CanvasLayer/Match_Control/Moves_Control/Moves_Container
 func update_move_icons():
-	move_icons.get_children()[2].visible = (max_moves == absolute_max_moves)
+	# move_icons.get_children()[2].visible = (max_moves == absolute_max_moves)
+	for move_icon in move_icons.get_children():
```

Comment:
This is better. Did you remove  absolute_max_moves or is it needed somewhere else?